src/utils/api-request.ts

```diff
@@ -505,7 +505,12 @@ class AsyncHttpCall {
       // Listen to timeouts and throw an error.
       req.setTimeout(timeout, timeoutCallback);
       req.on('socket', (socket) => {
+        socket.setMaxListeners(socket.getMaxListeners() + 1);
         socket.setTimeout(timeout, timeoutCallback);
+        socket.on('end', () => {
+          socket.setTimeout(0);
+          socket.setMaxListeners(socket.getMaxListeners() - 1);
```

Comment:
Can this be a negative value? If so, should we add `Math.max(emitter.getMaxListeners() - 1, 0)` to make sure we don't set this to a negative value?

Reply:
make sense 👍  added `Math.max`